day_generator.dart

```diff
@@ -105,3 +115,69 @@ class Day$dayNumber extends GenericDay {
 
 ''';
 }
+
+String _testTemplate(String day) {
+  return '''
+import 'package:test/test.dart';
+
+import '../solutions/day$day.dart';
+
+// *******************************************************************
+// Fill out the variables below according to the puzzle description!
+// The test code should usually not need to be changed.
+// *******************************************************************
+
+/// Paste in the small example that is given for the FIRST PART of the puzzle.
+/// It will be evaluated again the `_exampleSolutionPart1` below.
+const _exampleInput1 = \'''
+\''';
+
+/// Paste in the small example that is given for the SECOND PART of the puzzle.
+/// It will be evaluated against the `_exampleSolutionPart2` below.
+///
+/// In case the second part uses the same example, uncomment below line instead:
+// const _exampleInput2 = _exampleInput;
+const _exampleInput2 = \'''
+\''';
+
+/// The solution for the FIRST PART's example, which is given by the puzzle.
+const _exampleSolutionPart1 = 0;
+
+/// The solution for the SECOND PART's example, which is given by the puzzle.
+const _exampleSolutionPart2 = 0;
+
+/// The actual solution for the FIRST PART of the puzzle, based on your input.
+/// This can only be filled out after you have solved the puzzle and is used
+/// for regression testing when refactoring.
+const _puzzleSolutionPart1 = 0;
+
+/// The actual solution for the SECOND PART of the puzzle, based on your input.
+/// This can only be filled out after you have solved the puzzle and is used
+/// for regression testing when refactoring.
+const _puzzleSolutionPart2 = 0;
+
+void main() {
+  group(
+    'Day $day - Example Input',
+    () {
+      test('Part 1', () {
+        final day = Day$day()..inputForTesting = _exampleInput1;
+        expect(day.solvePart1(), _exampleSolutionPart1);
+      });
+      test('Part 2', () {
+        final day = Day$day()..inputForTesting = _exampleInput2;
+        expect(day.solvePart2(), _exampleSolutionPart2);
+      });
+    },
+  );
+  group(
+    'Day $day - Puzzle Input',
+    () {
+      final day = Day$day();
+      test('Part 1', () => expect(day.solvePart1(), _puzzleSolutionPart1));
+      test('Part 2', () => expect(day.solvePart2(), _puzzleSolutionPart2));
+    },
+  );
```

Comment:
Should we keep the `Puzzle Input` group commented out by default to prevent the tests from failing?

Reply:
Else the test suite would still fail until you provide the `_puzzleSolutionPart1` and `_puzzleSolutionPart2` variables and make it less clear that your solution worked.

Reply:
That's a good point! Pushed the changes

Reply:
We could be a bit more "smart" about it and skip the test if the solution hasn't been filled in:
```
const _puzzleSolutionPart1 = null;
const _puzzleSolutionPart2 = null;

group(
    'Day 01 - Puzzle Input',
    () {
      final day = Day01();
      test(
        'Part 1',
        skip: _puzzleSolutionPart1 == null
            ? 'Skipped because _puzzleSolutionPart1 is null'
            : false,
        () => expect(day.solvePart1(), _puzzleSolutionPart1),
      );
      test(
        'Part 1',
        skip: _puzzleSolutionPart2 == null
            ? 'Skipped because _puzzleSolutionPart2 is null'
            : false,
        () => expect(day.solvePart1(), _puzzleSolutionPart2),
      );
    },
  );
  ```

Reply:
That is perfect! I will add and commit it! 🚀
Was also thinking about skipping the tests without touching the actual test code, but stopped at the thought of a `shouldTest` boolean set by the user 😅
Thanks a lot! 💙